pkg/resource/deploy/plan_executor.go

```diff
@@ -60,15 +60,15 @@ func createTargetMap(targets []resource.URN) map[resource.URN]bool {
 // checkTargets validates that all the targets passed in refer to existing resources.  Diagnostics
 // are generated for any target that cannot be found.  The target must either have existed in the stack
 // prior to running the operation, or it must be the urn for a resource that was created.
-func (pe *planExecutor) checkTargets(targets []resource.URN) result.Result {
+func (pe *planExecutor) checkTargets(targets []resource.URN, op StepOp) result.Result {
 	if len(targets) == 0 {
 		return nil
 	}
 
 	olds := pe.plan.olds
 	var news map[resource.URN]bool
-	if pe.stepGen != nil && pe.stepGen.creates != nil {
-		news = pe.stepGen.creates
+	if pe.stepGen != nil {
+		news = pe.stepGen.urns
```

Comment:
This fixes a bug that caused a "resource not found" error if a specified target was replaced rather than created or updated.